public/components/getting_started/home.tsx

```diff
@@ -3,45 +3,107 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import { HashRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';
-import { TraceAnalyticsCoreDeps } from '../trace_analytics/home';
 import {
   ChromeBreadcrumb,
   ChromeStart,
   HttpStart,
+  MountPoint,
   NotificationsStart,
+  SavedObjectsStart,
 } from '../../../../../src/core/public';
+import {
+  DataSourceManagementPluginSetup,
+  DataSourceSelectableConfig,
+} from '../../../../../src/plugins/data_source_management/public';
 import { NewGettingStarted } from './components/getting_started';
-
-export type AppAnalyticsCoreDeps = TraceAnalyticsCoreDeps;
+import { dataSourceFilterFn } from '../../../common/utils/shared';
 
 export interface HomeProps extends RouteComponentProps {
   pplService: any;
   parentBreadcrumb: ChromeBreadcrumb;
   http: HttpStart;
   chrome: ChromeStart;
   notifications: NotificationsStart;
+  dataSourceEnabled: boolean;
+  dataSourceManagement: DataSourceManagementPluginSetup;
+  savedObjectsMDSClient: SavedObjectsStart;
+  setActionMenu: (menuMount: MountPoint | undefined) => void;
 }
 
 export const Home = (props: HomeProps) => {
-  const { http, chrome, pplService, notifications } = props;
+  const {
+    http,
+    chrome,
+    pplService,
+    notifications,
+    dataSourceEnabled,
+    dataSourceManagement,
+    savedObjectsMDSClient,
+    parentBreadcrumb,
+    setActionMenu,
+  } = props;
+
+  const [selectedDataSourceId, setSelectedDataSourceId] = useState<string>('');
+  const [selectedDataSourceLabel, setSelectedDataSourceLabel] = useState<string>('');
+
+  const onSelectedDataSourceChange = (e: any) => {
+    const dataSourceId = e[0] ? e[0].id : undefined;
+    const dataSourceLabel = e[0] ? e[0].label : '';
+    setSelectedDataSourceId(dataSourceId);
+    setSelectedDataSourceLabel(dataSourceLabel);
+  };
+
+  const DataSourceMenu = useMemo(() => {
+    if (dataSourceEnabled && dataSourceManagement?.ui) {
+      return dataSourceManagement.ui.getDataSourceMenu<DataSourceSelectableConfig>();
+    }
+    return null;
+  }, [dataSourceManagement, dataSourceEnabled]);
+
+  const dataSourceMenuComponent = useMemo(() => {
```

Comment:
That's a good suggestion, this calls for a higher level refactor rather than the scope of this backport PR. May be we can track this separately. Prematurely thinking, we may just have to create a react component wrapper over the components exposed by DSM plugin. Moreover, if we create this to be more generic, might as well end up creating a very similar thing and that is not advised.